src/wp2hugo/internal/hugogenerator/hugopage/wordpress_gallery_converter.go

```diff
@@ -0,0 +1,87 @@
+package hugopage
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+)
+
+// Example:  [gallery size="medium" link="file" columns="4" ids="1710,1713,1712,1711"]
+// Full examples : https://codex.wordpress.org/Gallery_Shortcode
+// the important field to extract is "ids", but others might come in handy
+// `size` is legacy from pre-responsive design and should be discarded now
+// `link` is probably something to enforce in Hugo figure shortcode,
+// for us it's mostly "file" to handle, since "attachment_page" makes no sense for Hugo.
+var _GalleryRegEx = regexp.MustCompile(`\[gallery ([^\[\]]*)\]`)
+
+var _idRegEx = regexp.MustCompile(`ids="([^"]+)"`)
+var _colsRegEx = regexp.MustCompile(`columns="([^"]+)"`)
+
+// TODO: should we handle `order="ASC|DESC"` when `orderby="ID"` ?
+// Seems to me that people mostly order pictures in galleries arbitrarily.
+
+// Converts the WordPress's caption shortcode to Hugo shortcode "figure"
+// https://adityatelange.github.io/hugo-PaperMod/posts/papermod/papermod-faq/#centering-image-in-markdown
+func replaceGalleryWithFigure(provider ImageURLProvider, htmlData string) string {
+	log.Debug().
+		Msg("Replacing gallery with figures")
+
+	htmlData = replaceAllStringSubmatchFunc(_GalleryRegEx, htmlData,
+		func(groups []string) string {
+			return galleryReplacementFunction(provider, groups)
+		})
+
+	return htmlData
+}
+
+func galleryReplacementFunction(provider ImageURLProvider, groups []string) string {
+	var output strings.Builder
+
+	// Find columns layout
+	cols := _colsRegEx.FindStringSubmatch(groups[1])
+	col_nb := "1"
+	if cols != nil {
+		col_nb = cols[1]
+	}
+
+	// Find image IDs
+	ids := _idRegEx.FindStringSubmatch(groups[1])
+	ids_array := strings.Split(ids[1], ",")
+
+	// TODO: maybe handle `order="ASC|DESC"` in conjunction with `orderby="..."`, so reorder ids_array here.
+
+	// We will use `figure` shortcodes nested into a `gallery` shortcode for the main layout
+	output.WriteString("<br>") // This will get converted to newline later on
+
+	// Opening tag :
+	output.WriteString(fmt.Sprintf(`{{%% gallery cols="%s" %%}}`,  col_nb))
```

Comment:
> 	output.WriteString(fmt.Sprintf(`{{%% gallery cols="%s" %%}}`,  col_nb))

Shouldn't this be

```
	output.WriteString(fmt.Sprintf(`{{< gallery cols="%s" >}}`,  col_nb))
```

Reply:
It's a little documented feature, but basically you need the `%%` syntax when nesting shortcodes. See for reference : https://discourse.gohugo.io/t/sharing-scratch-between-shortcode-outer-and-inner/48968/15?u=aurelienpierre

Reply:
Was it working for you?
It didn't work for me and then I changed `{{%` -> `{{<` manually and the generated code did show me the gallery.

Reply:
Yes, it works here. What version of Hugo are you using ? Also, it needs the config file to allow unsafe HTML parsing in Golmark renderer.

Reply:
I changed the syntax @ashishb, it works too here (because `.Inner` is explicitly parsed in the shortcode anyway). I don't know why the `{{%` syntax was needed at some point.